File: schemas/user-profile.ts
/**
 * Firestore path: users/{uid}
 * Mirror: ios/Shared/UserProfile.swift
 */
export interface UserProfile {
  displayName?: string;
  email?: string;
  yearLevel?: 'Y7' | 'Y8' | 'Y9' | 'Y10' | 'Y11' | 'Y12';
  fitnessLevel?: 'basic' | 'average' | 'intense';
  activePlanId?: string | null;
  totalXp?: number;
  isCoach?: boolean;
  teamId?: string | null;
  teamName?: string;
  health?: {
    latestHr?: number;
    latestSteps?: number;
    latestSleep?: number;
    restingHr?: number;
    lastSync?: string;
  };
  createdAt?: FirestoreTimestamp;
}

export type FirestoreTimestamp =
  | { toDate(): Date }
  | { seconds: number; nanoseconds: number }
  | string;

File: ios/Shared/UserProfile.swift
import Foundation

struct UserProfile: Identifiable, Equatable {
    let id: String
    var email: String?
    var displayName: String?
    var yearLevel: String?
    var fitnessLevel: String?
    var totalXp: Int?
    var activePlanId: String?

    init(from data: [String: Any], id: String) {
        self.id = id
        self.email = data["email"] as? String
        self.displayName = data["displayName"] as? String
        self.yearLevel = data["yearLevel"] as? String
        self.fitnessLevel = data["fitnessLevel"] as? String
        self.totalXp = data["totalXp"] as? Int
        self.activePlanId = data["activePlanId"] as? String
    }

    init(id: String) {
        self.id = id
    }
}
